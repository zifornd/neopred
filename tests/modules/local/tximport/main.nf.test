nextflow_process {

    name "Test Process TXIMPORT"
    script "modules/local/tximport/main.nf"
    process "TXIMPORT"

    test("Should run without failures") {

        when {
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
            }
            process {
                """
                // define inputs of the process here. Example:
                input[0] = Channel.fromPath("$projectDir/tests/test_result/tx2gene/quants/test")
                input[1] = Channel.fromPath("$projectDir/tests/test_result/tx2gene.tsv")
                input[2] = params.quant_type

                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
