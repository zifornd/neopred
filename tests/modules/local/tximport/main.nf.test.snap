{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "salmon.merged.gene_tpm.tsv:md5,5cf1f0e6bac2b77a9044dcd2969f0327"
                ],
                "1": [
                    "salmon.merged.gene_counts.tsv:md5,c643a2b6a54b8d13779428a49440e7a6"
                ],
                "2": [
                    "salmon.merged.gene_counts_length_scaled.tsv:md5,17da3afa445598ef99fd22e1f27de10e"
                ],
                "3": [
                    "salmon.merged.gene_counts_scaled.tsv:md5,6c7f05e42b7cb239deefa9c8e8d9aedf"
                ],
                "4": [
                    "salmon.merged.gene_lengths.tsv:md5,a4fad3e7437c9d7b50513bd8c2acd08d"
                ],
                "5": [
                    "salmon.merged.transcript_tpm.tsv:md5,ff95f618962a667a16d79d7c75babc33"
                ],
                "6": [
                    "salmon.merged.transcript_counts.tsv:md5,523fd6f6838e17357e4160f1d9279fb7"
                ],
                "7": [
                    "salmon.merged.transcript_lengths.tsv:md5,408f7cab060f574929895c94f85253bd"
                ],
                "8": [
                    "versions.yml:md5,367f68930d7a2fbf18a3416ef4d6c2b1"
                ],
                "counts_gene": [
                    "salmon.merged.gene_counts.tsv:md5,c643a2b6a54b8d13779428a49440e7a6"
                ],
                "counts_gene_length_scaled": [
                    "salmon.merged.gene_counts_length_scaled.tsv:md5,17da3afa445598ef99fd22e1f27de10e"
                ],
                "counts_gene_scaled": [
                    "salmon.merged.gene_counts_scaled.tsv:md5,6c7f05e42b7cb239deefa9c8e8d9aedf"
                ],
                "counts_transcript": [
                    "salmon.merged.transcript_counts.tsv:md5,523fd6f6838e17357e4160f1d9279fb7"
                ],
                "lengths_gene": [
                    "salmon.merged.gene_lengths.tsv:md5,a4fad3e7437c9d7b50513bd8c2acd08d"
                ],
                "lengths_transcript": [
                    "salmon.merged.transcript_lengths.tsv:md5,408f7cab060f574929895c94f85253bd"
                ],
                "tpm_gene": [
                    "salmon.merged.gene_tpm.tsv:md5,5cf1f0e6bac2b77a9044dcd2969f0327"
                ],
                "tpm_transcript": [
                    "salmon.merged.transcript_tpm.tsv:md5,ff95f618962a667a16d79d7c75babc33"
                ],
                "versions": [
                    "versions.yml:md5,367f68930d7a2fbf18a3416ef4d6c2b1"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.1"
        },
        "timestamp": "2025-01-08T06:18:29.523462"
    }
}